src/index.ts

```diff
@@ -0,0 +1,104 @@
+import { IFetchComponent } from "@well-known-components/http-server"
+import { IConfigComponent, ILoggerComponent, IMetricsComponent } from "@well-known-components/interfaces"
+import { ISubgraphComponent, SubgraphResponse, Variables } from "./types"
+import { sleep } from "./utils"
+
+/**
+ * @public
+ */
+export async function createSubgraphComponent(
+  url: string,
+  components: createSubgraphComponent.NeededComponents
+): Promise<ISubgraphComponent> {
+  const { logs, metrics, config, fetch } = components
+
+  const logger = logs.getLogger("thegraph-port")
+
+  const RETRIES = (await config.getNumber("SUBGRAPH_COMPONENT_RETRIES")) || 3
+  const TIMEOUT = (await config.getNumber("SUBGRAPH_COMPONENT_QUERY_TIMEOUT")) || 5000
+  const BACKOFF = (await config.getNumber("SUBGRAPH_COMPONENT_BACKOFF")) || 500
+
+  async function executeQuery<T>(query: string, variables: Variables = {}, remainingAttempts?: number): Promise<T> {
```

Comment:
This could replace the line 23.
```suggestion
  async function executeQuery<T>(query: string, variables: Variables = {}, remainingAttempts?: number = RETRIES): Promise<T> {
```